### ISSUE TITLE
ACA-Py - AFJ: DID Ex Req fails: did:peer:1 does not correspond to a sovrin DID

### ISSUE THREAD

**Issue by dbluhm** [hyperledger/aries-cloudagent-python#2610]
Similar exchange to #2608; this occurs after applying the fix in #2609.

```python
2023-11-17 02:34:21,812 aries_cloudagent.core.dispatcher ERROR Handler error: Dispatcher.handle_message
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/asyncio/tasks.py", line 256, in __step
    result = coro.send(None)
  File "/aries_cloudagent/core/dispatcher.py", line 253, in handle_message
    await handler(context, responder)
  File "/aries_cloudagent/protocols/didexchange/v1_0/handlers/request_handler.py", line 39, in handle
    conn_rec = await mgr.receive_request(
  File "/aries_cloudagent/protocols/didexchange/v1_0/manager.py", line 486, in receive_request
    conn_did_doc = await self.verify_diddoc(wallet, request.did_doc_attach)
  File "/aries_cloudagent/protocols/didexchange/v1_0/manager.py", line 943, in verify_diddoc
    return DIDDoc.deserialize(json.loads(signed_diddoc_bytes.decode()))
  File "/aries_cloudagent/connections/models/diddoc/diddoc.py", line 248, in deserialize
    rv = DIDDoc(did_doc["id"])
  File "/aries_cloudagent/connections/models/diddoc/diddoc.py", line 57, in __init__
    self._did = canon_did(did) if did else None  # allow specification post-hoc
  File "/aries_cloudagent/connections/models/diddoc/util.py", line 57, in canon_did
    raise ValueError(
ValueError: Bad specification did:peer:1zQmT5FapKZqTdnej34vvg9ys98T2FiC8JWDRFE8rEFxP9PJ does not correspond to a sovrin DID
```

**Comment by dbluhm** [comment]
I see a fix for this going one of two ways:
- We find a quick way to patch the current DID Ex handling to permit did:peer:1 (we don't actually do much with the DID value anyways)
- We forget about it and focus on pushing did:peer:2/4

**Comment by dbluhm** [comment]
Fixed by #2611
